src/components/Identity/injectAddQueryStringIdentityToPayload.js

```diff
@@ -0,0 +1,42 @@
+// Example: adobe_mc=TS%3D1641432103%7CMCMID%3D77094828402023918047117570965393734545%7CMCORGID%3DFAF554945B90342F0A495E2C%40AdobeOrg
+// Decoded: adobe_mc=TS=1641432103|MCMID=77094828402023918047117570965393734545|MCORGID=FAF554945B90342F0A495E2C@AdobeOrg
+
+import { queryString } from "../../utils";
+import queryStringIdentityParam from "../../constants/queryStringIdentityParam";
+import ecidNamespace from "../../constants/ecidNamespace";
+
+export default ({ locationSearch, dateProvider, orgId, logger }) => payload => {
+  if (payload.hasIdentity(ecidNamespace)) {
+    // don't overwrite a user provided ecid identity
+    return;
+  }
+
+  const parsedQueryString = queryString.parse(locationSearch);
+  const queryStringValue = parsedQueryString[queryStringIdentityParam];
+  if (queryStringValue === undefined) {
+    return;
+  }
+
+  const properties = queryStringValue.split("|").reduce((memo, keyValue) => {
+    const [key, value] = keyValue.split("=");
+    memo[key] = value;
+    return memo;
+  }, {});
+  const ts = parseInt(properties.TS, 10);
+  const mcmid = properties.MCMID;
+  const mcorgid = properties.MCORGID;
+
+  // all inequalities with NaN variables are false in javascript
+  if (
+    dateProvider().getTime() / 1000 <= ts + 300 &&
+    mcorgid === orgId &&
+    mcmid
+  ) {
+    logger.info(`Adding ECID identity ${mcmid} to identityMap.`);
```

Comment:
Should we mention the source of this ECID? Like we're getting it from adobe_mc? It'd be symmetrical with the exception message below.